src/base.jl

```diff
@@ -270,26 +281,61 @@ Serializes a system to a JSON string.
 - `filename::AbstractString`: filename to write
 
 # Keyword arguments
+- `user_data::Union{Nothing, Dict} = nothing`: optional metadata to record
+- `pretty::Bool = false`: whether to pretty-print the JSON
 - `force::Bool = false`: whether to overwrite existing files
 - `check::Bool = false`: whether to run system validation checks
 
 Refer to [`check_component`](@ref) for exceptions thrown if `check = true`.
 """
-function IS.to_json(sys::System, filename::AbstractString; force = false, runchecks = false)
+function IS.to_json(
+    sys::System,
+    filename::AbstractString;
+    user_data = nothing,
+    pretty = false,
+    force = false,
+    runchecks = false,
+)
     if runchecks
         check(sys)
         check_components(sys)
     end
 
     IS.prepare_for_serialization!(sys.data, filename; force = force)
-    data = to_json(sys)
+    data = to_json(sys; pretty = pretty)
     open(filename, "w") do io
         write(io, data)
     end
+
+    mfile = joinpath(dirname(filename), splitext(basename(filename))[1] * "_metadata.json")
     @info "Serialized System to $filename"
+    _serialize_system_metadata_to_file(sys, mfile, user_data)
     return
 end
 
+function _serialize_system_metadata_to_file(sys::System, filename, user_data)
+    name = get_name(sys)
+    description = get_description(sys)
+    resolution = Dates.Minute(get_time_series_resolution(sys)).value
+    metadata = OrderedDict(
+        "name" => isnothing(name) ? "" : name,
+        "description" => isnothing(description) ? "" : description,
+        "frequency" => sys.frequency,
+        "time_series_resolution_minutes" => resolution,
```

Comment:
This reports the resolution as an integer and the units are in the field name. An alternative is like this:
```
"time_series_resolution": "60 minutes"
```
The string is probably easier to read. The integer is much easier to machine parse.